crates/store/src/db/models/queries.rs

```diff
@@ -0,0 +1,1647 @@
+                .get_results::<Vec<u8>>(conn)?;
+        let accounts = deserialize_raw_vec::<_, Account>(account_details_serialized)?;
+        Ok(accounts)
+    }
+
+    let mut count = 0;
+    for update in accounts {
+        let account_id = update.account_id();
+        // Extract the 30-bit prefix to provide easy look ups for NTB
+        // Do not store prefix for accounts that are not network
+        let network_account_id_prefix = if account_id.is_network() {
+            Some(NetworkAccountPrefix::try_from(account_id)?)
+        } else {
+            None
+        };
+
+        let (full_account, insert_delta) = match update.details() {
+            AccountUpdateDetails::Private => (None, None),
+            AccountUpdateDetails::New(account) => {
+                debug_assert_eq!(account_id, account.id());
+
+                if account.commitment() != update.final_state_commitment() {
+                    return Err(DatabaseError::AccountCommitmentsMismatch {
+                        calculated: account.commitment(),
+                        expected: update.final_state_commitment(),
+                    });
+                }
+
+                let insert_delta = build_insert_delta(account)?;
+
+                (Some(Cow::Borrowed(account)), Some(Cow::Owned(insert_delta)))
+            },
+            AccountUpdateDetails::Delta(delta) => {
+                let mut rows = select_details_stmt(conn, account_id)?.into_iter();
+                let Some(account) = rows.next() else {
+                    return Err(DatabaseError::AccountNotFoundInDb(account_id));
+                };
+
+                let account = apply_delta(account, delta, &update.final_state_commitment())?;
+
+                (Some(Cow::Owned(account)), Some(Cow::Borrowed(delta)))
+            },
+        };
+
+        let val = (
+            schema::accounts::account_id.eq(account_id.to_bytes()),
+            schema::accounts::network_account_id_prefix
+                .eq(network_account_id_prefix.map(network_account_prefix_to_raw_sql)),
+            schema::accounts::account_commitment.eq(update.final_state_commitment().to_bytes()),
+            schema::accounts::block_num.eq(block_number_to_raw_sql(block_num)),
+            schema::accounts::details.eq(full_account.as_ref().map(|account| account.to_bytes())),
+        );
+        let v = val.clone();
+        let inserted = diesel::insert_into(schema::accounts::table)
+            .values(&v)
+            .on_conflict(schema::accounts::account_id)
+            .do_update()
+            .set(val)
+            .execute(conn)?;
+
+        debug_assert_eq!(inserted, 1);
+
+        if let Some(delta) = insert_delta {
+            insert_account_delta(conn, account_id, block_num, &delta)?;
+        }
+
+        count += inserted;
+    }
+
+    Ok(count)
+}
+
+/// Insert notes to the DB using the given [Transaction]. Public notes should also have a nullifier.
```

Comment:
Similar comment as above but about `[Transaction]` - it should probably be `SqliteConnection` or just `Connection` if we alias it.

Also, to confirm `SqliteConnection` actually works like a transaction, right? i.e., multiple updates made against the same `SqliteConnection` will be committed atomically.

Reply:
> I've mentioned in some comment elsewhere that's lost in the stack that I'd like a follow-up that forces `Transaction` everywhere, probably by creating our own wrapper type.

Let's create an issue for this (unless one already exists). I think it would be great to be able to tell by looking at the code which function expect a simple connection and which ones expect a transaction (unless all of them expect a transaction?).

Reply:
We do have `fn transact` and `fn query` - if in doubt, we might just remove `query` and enforce a transaction and avoid additional complexity. We only have a handful of queries that don't have transactions.

Reply:
I think the point is more that the functions in `query.rs` should _not_ be allowing anything that isn't a transaction.

But I would also remove `query` because imo _all_ business logic should be using a transaction.